panelapp/api/v1/viewsets.py

```diff
@@ -197,6 +204,123 @@ class EntityViewSet(viewsets.mixins.ListModelMixin, viewsets.GenericViewSet):
     lookup_field = "entity_name"
     lookup_url_kwarg = "entity_name"
 
+    def filter_list(self, object):
```

Comment:
Since there are many `if else` statements which have the same logic, could you rewrite this method so each step is evaluated just once, for example:

```
filters = []
if entity_name:
    filters.append(object["entity_name"] in entity_name.split(","))
if conf _level:
    filters.append(object["confidence_level"] == confidence_level)
if tags:
    ..etc

if filters:
     return all(filters)
else:
    return True
```